server.go

```diff
@@ -28,11 +29,23 @@ func (o ServerOptions) TLSAddr() string {
 	return fmt.Sprintf("%s:%d", o.listeningAddr, o.listeningTLSPort)
 }
 
+func (o ServerOptions) ClientsAddr() string {
+	return fmt.Sprintf("%s:%d", o.listeningAddr, o.listeningForClientsPort)
+}
+func NewServerOptions(listeningAddr string, listeningTLSPort, listeningHTTPPort, listeningForClientsPort uint) ServerOptions {
```

Comment:
I would get rid of this one. All field are public anyhow so you can just create the object using a literal.